utils.py

```diff
@@ -597,6 +728,21 @@ def file_exists(file_path):
     else:
         return False
 
+def check_file_exists(file_path):
+    if not os.path.exists(file_path):
+        raise argparse.ArgumentTypeError(f"Invalid file path: '{file_path}'. File does not exist.")
+    return (file_path)
+
+def check_file_type(file_path, file_type):
+    if not file_path.endswith(f".{file_type}"):
+        raise argparse.ArgumentTypeError(f"Invalid file type: '{file_path}'. Must be of type '{file_type}.'")
+    return file_path
+
+def check_appimage(file_path):
+    file_path = check_file_exists(file_path)
+    file_path = check_file_type(file_path, "AppImage")
+    return file_path
```

Comment:
I think `check_appimage()` could use `file_exists()` also defined here in `utils.py` rather than adding `check_file_exists()`. It would need to handle a `False` result.

In general, we've tended to handle incompatible variable values by returning `False` rather than raising an exception, both of which need to then be handled anyway. Any reason not to just return `False` in `check_file_type()`?